File: Cpp/v1/testFunctions.cc
#include <iostream>
#include "polynomial.h"

int main(void){
  std::cout << "Testing the polynomial class" << std::endl;
  
  std::vector<double> coeffs1 = {1,2,3};
  std::vector<double> coeffs2 = {100,200};
  Polynomial *p1 = new Polynomial(coeffs1);
  Polynomial *p2 = new Polynomial(coeffs2);
  
  std::cout << "Polynomial p1:" << std::endl;
  p1->printPolynomial();
  std::cout << "Polynomial p2:" << std::endl;
  p2->printPolynomial();
  
  std::cout << "p1.add(p2) gives:" << std::endl;
  (p1.add(p2))->printPolynomial();
  
  std::cout << "p1+p2 gives:" << std::endl;
  (p1+p2)->printPolynomial();

  std::cout << "p1(1) = " << p1(1) << std::endl;
  std::cout << "p1(-1) = " << p1(-1) << std::endl;
  
  std::cout << "(p1+p2)(10) = " << (p1+p2)(10) << std::endl;
  
  std::cout << "p1.mul(p1) gives" << std::endl;
  (p1->mul(p1))->printPolynomial();
  
  std::cout << "p1 * p1 gives" << std::endl;
  (p1 * p1)->printPolynomial();

  std::cout << "p1 * p2 + p1 gives" << std::endl;
  (p1*p2 + p1)->printPolynomial();

  std::cout << "p1.roots() gives:" << std::endl;
  printVec(p1->roots());
  
  std::cout << "p2.roots() gives:" << std::enld;
  printVec(p2->roots());
  
  std::cout << "Polynomial p3:" << std::endl;

  std::vector<double> coeffs3 = {3,2,-1};
  Polynomial *p3 = new Polynomial(coeffs3);
  p3->printPolynomial();
  
  std::cout << "p3.roots() gives:" << std::endl;
  printVec(p3->roots());
  
  strd::cout << "p1*p2.roots() gives:" << std::endl;
  printVec((p1*p2)->roots());
  
 
  return 0;
}

File: Python/testFunctions.py
import os,sys

from polynomial import polynomial

def main():

    print 'Creating Polynomial p1 with list [1,2,3]'
    p1 = polynomial([1,2,3])
    'Representation of p1 gives:'
    print p1
    
    print 'Creating polynomial p2 with list [100,200]'
    p2 = polynomial([100,200])
    
    print 'p1.add(p2) gives'
    print p1.add(p2)
    print 'p1+p2 gives'
    print p1+p2

    print 'p1(1) gives'
    print p1(1)
    
    print 'p1(-1) gives'
    print p1(-1)

    print '(p1+p2)(10) gives'
    print (p1+p2)(10)

    print 'p1.mul(p1) gives'
    print p1.mul(p1)
    print 'p1 * p1 gives'
    print p1*p1
    print 'p1*p2 + p1 gives'
    print p1 * p2 + p1
    
    print 'p1.roots gives'
    print p1.roots()
    print 'p2 roots gives'
    print p2.roots()

    print 'make polynomial 3 with list [3,2,-1]'
    p3 = polynomial([3,2,-1])
    print 'p3.roots is'
    print p3.roots()

    print 'p1*p2 roots gives:'
    print (p1*p2).roots()


if __name__ == "__main__":
    main()
